p2p/chain.py

```diff
@@ -443,9 +410,7 @@ def _handle_block_headers(self, headers: List[BlockHeader]) -> None:
         self.logger.debug("Got Receipts for %d blocks from %s", len(receipts_by_block), peer)
         loop = asyncio.get_event_loop()
         iterator = map(make_trie_root_and_nodes, receipts_by_block)
-        receipts_tries = await wait_with_token(
-            loop.run_in_executor(self._executor, list, iterator),
-            token=self.cancel_token)
+        receipts_tries = await self.wait_first(loop.run_in_executor(self._executor, list, iterator))
```

Comment:
Can we add a comment explaining what is happening here.  It makes sense to me because I know `map` is lazy and thus, the `list` application pushes the evaluation down into the executor.  Not sure how clear that will be to others reading the codebase.

Reply:
Sure, that is indeed not clear